0000-product.md

```diff
@@ -0,0 +1,463 @@
+  state.
+* ProductDelete - remove a Product from state.
+
+## Permissions
+
+Creation of GS1 products is restricted to agents acting on behalf of the
+organization that is associated with the GS1 company prefix encoded in the GTIN.
+An organization must have the GS1 company prefix in the “gs1_company_prefixes”
+metadata field for its Pike organization.  (Organization-level metadata will
+need to be implemented in Pike.) When a product is created, its owning
+organization is stored with the product in an “owner” field.
+
+Updates to products is restricted to agents acting on behalf of the organization
+stored in the product’s owner field.  Only property fields can be updated.
+Product_id, product_namespace, and owner fields are immutable.
+
+Deletion of products is restricted to agents acting on behalf of the
+organization in the product’s owner field.  A setting will turn off deletion
+entirely, with the intent that it could be enabled/disabled by a Grid
+administrator.  Disabling delete is useful because external systems may have
+references to these products and deleting them could leave dangling references.
+
+## GTIN Validation
+
+Creation of a resuable gtin validation function to programmatically express the
+equation used to validate a GTIN. It validates gtin format to avoid mistype
+errors similar to a credit card validation. It's implemented as an extensible
+function such that further validation steps can be added, if needed. For details
+on the equation see: [Check digit
+validation](https://www.gs1.org/services/how-calculate-check-digit-manually).
+
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## State
+
+### Product Representation
+
+The primary object stored in state is “Product”, which consists of an product_id
+(GS1 GTIN), an owner (org_id compatible w/Pike), and a list of property
+name/value pairs.  The properties available are defined by the Grid Property
+Schema Transaction Family and are restricted to the fields and rules of the GS1
+Product schema.  Transactions which are responsible for setting product state
+values must ensure that the properties conform with the requirements of the GS1
+Product Property Schema.
+
+``` 
+message Product { 
+    enum ProductNamespace { 
+        UNSET_NAMESPACE = 0; 
+        GS1 = 1; 
+    }
+    string product_id = 1; 
+    ProductNamespace product_namespace = 2; 
+    string owner = 3;
+    repeated PropertyValue properties = 4; 
+} 
+```
+
+The GS1 GTIN is an identifier (product_id) used to identify trade items.  A GTIN
+is the data transmitted from a barcode scan and is made up of a company prefix
+(or GS1-8 prefix) and item reference.  We will initially support GTIN-12,
+GTIN-13, and GTIN-14. (GTIN-8 may be supported in the future.) The GS1 GTIN
+specification is documented in section 3.3.2, Identification of a trade item
+(GTIN): AI (01), on page 140 of the **GS1 General Specification**:
+
```

Comment:
So are we limiting this to barcodes and single Class level items for the near term, with plans to expand to other data carriers, levels of product hierarchy and assets in the long term?

Reply:
Correct

Reply:
use "data carrier" over "barcode"